src/main/java/com/cisco/trex/stateful/api/lowlevel/ASTFProfile.java

```diff
@@ -48,46 +47,100 @@ public ASTFProfile(
       ASTFGlobalInfo astfServerGlobalInfo,
       List<ASTFTemplate> astfTemplateList,
       List<ASTFCapInfo> astfCapInfoList) {
+    this(
+        defaultIpGen,
+        astfClientGlobalInfo,
+        astfServerGlobalInfo,
+        astfTemplateList,
+        astfCapInfoList,
+        null,
+        null);
+  }
+
+  /**
+   * constructor Define a ASTF profile You should give at least a template or a cap_list, maybe
+   * both.
+   *
+   * @param defaultIpGen ASTFIPGen
+   * @param astfClientGlobalInfo ASTFGlobalInfo tcp parameters to be used for client side, if
+   *     cap_list is given. This is optional. If not specified,TCP parameters for each flow will be
+   *     taken from its cap file.
+   * @param astfServerGlobalInfo Same as default_tcp_server_info for client side.
+   * @param astfTemplateList define a list of manual templates or one template
+   * @param astfCapInfoList define a list of pcap files list in case there is no templates
+   * @param sDelay ASTFCmdDelay or ASTFCmdDelayRnd see
+   *     :class:`trex.astf.trex_astf_profile.ASTFCmdDelay` and
+   *     :class:`trex.astf.trex_astf_profile.ASTFCmdDelayRnd` Server delay command before sending
+   *     response back to client. This will be applied on all cap in cap list, unless cap specified
+   *     his own s_delay. defaults to None means no delay.
+   * @param udpMtu int or None MTU for udp packets, if packets exceeding the specified value they
+   *     will be cut down from L7 in order to fit. This will be applied on all cap in cap list,
+   *     unless cap specified his own udp_mtu. defaults to None.
+   */
+  public ASTFProfile(
+      ASTFIpGen defaultIpGen,
+      ASTFGlobalInfo astfClientGlobalInfo,
+      ASTFGlobalInfo astfServerGlobalInfo,
+      List<ASTFTemplate> astfTemplateList,
+      List<ASTFCapInfo> astfCapInfoList,
+      ASTFCmd sDelay,
+      Integer udpMtu) {
+    if (!(sDelay instanceof ASTFCmdDelay) && !(sDelay instanceof ASTFCmdDelayRnd)) {
+      throw new IllegalStateException(
+          "bad param sDelay, it should be instanceof ASTFCmdDelayRnd or  ASTFCmdDelay");
+    }
```

Comment:
Instead of having this exception thrown, it's better create two constructors, one for ASTFCmdDelay, another for ASTFCmdDeplayRnd